### DISCUSSION TITLE
`envvar` not picked up in AWS Lambda context

### DISCUSSION THREAD

**Discussion by zopyx** [fastapi/typer#1328] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
database: str = typer.Option(
        ...,
        help="URL for Postgres sync database (e.g. postgresqlext://user:pw@localhost/mydb)",
        envvar="SYNC_DATABASE",
    )
```

### Description

I havr a `main()` method with a parameter `database` as documented above. It will pick up its configuration from the given environment variable `SYNC_DATABASE`. This works perfectly from the commandline.

Now, I want to use the same application code with `main()` as entry point from within an AWS Lambda function.
The related environment variable is correctly set through the AWS Lambda environment (checked by using logger messages).
So `SYNC_DATABASE` is correctly available for the Python process.

The Lambda handler invokes `typer.run(main)`.

However, the `database` parameter is always set to `OptionInfo`  rather than the value of the configured environment variable.

Expected behavior: `database`  should have the value of the env var `SYNC_DATABASE`.

### Operating System

Linux

### Operating System Details

AWS Lambda

### Typer Version

0.4.2

### Python Version

3.9

**Comment by zarch** [comment]
@zopyx did you export your variables?

I have a similar issue that I think is related to this.
I'm on debian testing, I'm in a dedicated and activated virtualenv.

The python code is:
```python
# file: test_typer.py
import typer
from typing_extensions import Annotated

arg_database: str = typer.Option(
        ...,
        help="URL for Postgres sync database (e.g. postgresqlext://user:pw@localhost/mydb)",
        envvar="SYNC_DATABASE",
    )


def main(database: Annotated[str, arg_database]):
    print(f"DATABASE: {database}")


if __name__ == "__main__":
    typer.run(main)
```

I prepared the environment in with the following commands:

```bash
🐍 v3.12.1 (.venv)
❯ echo "SYNC_DATABASE='my_nice_db'\n" > myenv

🐍 v3.12.1 (.venv)
❯ cat myenv
SYNC_DATABASE='my_nice_db'

🐍 v3.12.1 (.venv)
❯ source myenv

syngw on  bus.nats [!+?⇡] is 📦 v0.0.1 via 🐍 v3.12.1 (.venv)
❯ echo $SYNC_DATABASE
my_nice_db
```

Now I tested the code with:

```bash
🐍 v3.12.1 (.venv)
❯ python test_typer.py --database="my_nice_db"
DATABASE: my_nice_db

🐍 v3.12.1 (.venv)
❯ echo $SYNC_DATABASE
my_nice_db

 🐍 v3.12.1 (.venv)
❯ python test_typer.py
Usage: test_typer.py [OPTIONS]
Try 'test_typer.py --help' for help.
╭─ Error ────────────────────────────────────────────────╮
│ Missing option '--database'.                           │
╰────────────────────────────────────────────────────────╯

🐍 v3.12.1 (.venv)
❯ SYNC_DATABASE=$SYNC_DATABASE python test_typer.py
DATABASE: my_nice_db
```

So from the above test the command seems to take the environment variable only when passed explicitly from the command line, probably this is due that the variables are available only at process level and are not pass through the processes. Exporting the variable fixed my issue!

```bash
🐍 v3.12.1 (.venv)
❯ export SYNC_DATABASE="my_nice_db"

 🐍 v3.12.1 (.venv)
❯ python test_typer.py
DATABASE: my_nice_db
```
I believe that this should fix also your issue.

**Comment by zopyx** [comment]
LOL...shall I remember what I did two years ago?
